x/epoching/keeper/hooks.go

```diff
@@ -68,7 +76,16 @@ func (h Hooks) BeforeValidatorSlashed(ctx context.Context, valAddr sdk.ValAddres
 
 	for _, threshold := range thresholds {
 		// if a certain threshold voting power is slashed in a single epoch, emit event and trigger hook
-		if float64(slashedVotingPower) < float64(totalVotingPower)*threshold && float64(totalVotingPower)*threshold <= float64(slashedVotingPower+thisVotingPower) {
+		// Use integer multiplication instead of floating point comparison
+		// totalVotingPower is multiplied by the threshold (totalVotingPower * threshold) and then compared against slashedVotingPower
+		// and (slashedVotingPower + thisVotingPower). So, by multiplying totalVotingPower * threshold.numerator
+		// and slashedVotingPower & (slashedVotingPower + thisVotingPower) by the threshold.denominator
+		// we get the same comparison.
+		left := slashedVotingPower * threshold.denominator
```

Comment:
given that we are dealing with decimals maybe lets use `LegacyDec` type for computations (imo 2 place precision will be enough here)